RELEASE.md

```diff
@@ -2,22 +2,59 @@
 
 [![hackmd-github-sync-badge](https://hackmd.io/sVOAyv6LTwiQllQUctxP1w/badge)](https://hackmd.io/sVOAyv6LTwiQllQUctxP1w)
 
-## Resolve Issues
+## Versioning Scheme
+Flyte components follow [SemVer conventions](https://semver.org/). In a nutshell, that means every component has a release version that follows this format: `MAJOR.MINOR.PATCH`. This includes the aggregated milestone release described below.
+
+Flyte hasn't reached `v1.0.0` version yet. This version indicates a stable public API. After releasing `v1.0.0` any backward incompatible changes require a `MAJOR` version bump. The community and the team, however, strives to maintain backward compatibility with the current public APIs as much as possible and gracefully deprecating APIs when absolutely required.
+
+Until Flyte reaches `v1.0.0`, we are maintaining the following definitions of versions `MAJOR.MINOR.PATCHbX`:
+
+1. **MAJOR** is pinned to `0`.
+1. **MINOR** is bumped on one of two occasions:
+    1. A backward incompatible change or,
+
+       The decision to whether to a particular change constitutes a breaking change or not is evaluated at the component level (e.g. flytekit, flyteadmin... etc.)
+    1. A major feature push
+
+       At the end of every quarter, the core maintainers will determine if, for each component, the accumlated commits warrant a major feature push.
+1. **PATCH** is bumped on every milestone (defined below)
+2. **bX** is bumped on every commit.
+
+## Milestones
```

Comment:
we should also update - https://docs.flyte.org/en/latest/community/roadmap.html to point to here